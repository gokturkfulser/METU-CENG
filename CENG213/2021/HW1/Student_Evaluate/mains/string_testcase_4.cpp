#include <iostream>

#include "String.h"
//#include "../String.h"

/*
 * Case 4 : Default constructor; various inserts; getLength and isEmpty; print.
 */
int main() {
    std::cout << "-> Creating a string with default constructor." << std::endl;

    String str;

    std::cout << "-> Appending and inserting some characters to the string." << std::endl;

    str.appendCharacter('e');
    str.appendCharacter('h');
    str.insertCharacter('m', 0);
    str.appendCharacter('b');
    str.insertCharacter('r', 2);
    str.appendCharacter('a');
    str.insertCharacter('a', 4);

    std::cout << "-> Printing the string." << std::endl;

    str.print(true);

    std::cout << "-> Calculating the length." << std::endl;

    std::cout << "length is " << str.getLength() << std::endl;

    std::cout << "-> Checking if the string is empty or not." << std::endl;

    if (str.isEmpty()) {
        std::cout << "string is empty" << std::endl;
    } else {
        std::cout << "string is not empty" << std::endl;
    }

    return 0;
}
