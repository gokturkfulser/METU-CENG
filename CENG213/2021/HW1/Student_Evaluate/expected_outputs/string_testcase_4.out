-> Creating a string with default constructor.
-> Appending and inserting some characters to the string.
-> Printing the string.
['m','e','r','h','a','b','a']
-> Calculating the length.
length is 7
-> Checking if the string is empty or not.
string is not empty
